utilities/rpp-unittests/HOST_NEW/Tensor_host_pkd3.cpp

```diff
@@ -556,246 +849,265 @@ int main(int argc, char **argv)
 
         for (int i = 0; i < ioBufferSize; i++)
         {
-            *inputi8Temp = (Rpp8s) (((Rpp32s) *inputTemp) - 128);
-            *inputi8_secondTemp = (Rpp8s) (((Rpp32s) *input_secondTemp) - 128);
+            *inputi8Temp = (Rpp8s)(((Rpp32s)*inputTemp) - 128);
+            *inputi8_secondTemp = (Rpp8s)(((Rpp32s)*input_secondTemp) - 128);
             inputTemp++;
             inputi8Temp++;
             input_secondTemp++;
             inputi8_secondTemp++;
         }
     }
 
-    // Run case-wise RPP API and measure time
+    // Run case-wise RPP API and measure time for Unit test
 
     rppHandle_t handle;
     rppCreateWithBatchSize(&handle, noOfImages);
-    clock_t start, end;
-    double start_omp, end_omp;
-    double cpu_time_used, omp_time_used;
+
+    double max_time_used = 0, min_time_used = 500, avg_time_used = 0;
 
     string test_case_name;
 
-    switch (test_case)
+    int count_loop;
+    if (test_type == 0)
```

Comment:
No need of another variable here. Use num_iterations passed directly and update the value of num_iterations based on unit test/ performance test before being passed to this function